lib/nostrum/api.ex

```diff
@@ -4399,20 +3711,20 @@ defmodule Nostrum.Api do
   @doc since: "0.7.0"
   @spec delete_guild_auto_moderation_rule(Guild.id(), AutoModerationRule.id()) :: {:ok} | error
   def delete_guild_auto_moderation_rule(guild_id, rule_id) do
-    request(:delete, Constants.guild_auto_moderation_rule(guild_id, rule_id))
+    Nostrum.Api.AutoModeration.delete_rule(guild_id, rule_id)
   end
 
   @spec maybe_add_reason(String.t() | nil) :: list()
-  defp maybe_add_reason(reason) do
+  def maybe_add_reason(reason) do
```

Comment:
Instead of making all these public functions on the `Nostrum.Api` module as they need to be called elsewhere, what about moving them to `Nostrum.Api.Adapter`? Since that one has `@moduledoc false`.